F/fastp/build_tarballs.jl

```diff
@@ -0,0 +1,58 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "fastp"
+version = v"0.23.3"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/intel/isa-l.git", "2bbce31943289d5696bcf2a433124c50928226a2"),
+    GitSource("https://github.com/ebiggers/libdeflate.git", "02dfa32da3ee3982c66278e714d2e21276dfb67b"),
+    GitSource("https://github.com/OpenGene/fastp.git", "ca559a71feed94e74ea449e7567d0506de48dea4"),
```

Comment:
Having multiple sources is a good indication you should rather have separate packages.  And we already have [libdeflate](https://github.com/JuliaPackaging/Yggdrasil/blob/dc00ba45dc8953eb72128b72b847963d94bf9405/L/libdeflate/build_tarballs.jl).

Reply:
I made another pull request for isa-l, will adjust the fastp builder to use the binary dependencies when libisal_jll is available.